src/clusterfuzz/_internal/bot/tasks/utasks/minimize_task.py

```diff
@@ -1237,18 +1261,27 @@ def _unset_dedup_flags():
   return output_file_path, crash_result
 
 
-def _update_crash_result(testcase, crash_result, command):
-  """Update testcase with crash result."""
+def _jsonify_crash_result(crash_result, command):
+  """Extract necessary data from CrashResult."""
   min_state = crash_result.get_symbolized_data()
   min_unsymbolized_crash_stacktrace = crash_result.get_stacktrace(
       symbolized=False)
   min_crash_stacktrace = utils.get_crash_stacktrace_output(
       command, min_state.crash_stacktrace, min_unsymbolized_crash_stacktrace)
-  testcase.crash_type = min_state.crash_type
-  testcase.crash_address = min_state.crash_address
-  testcase.crash_state = min_state.crash_state
-  testcase.crash_stacktrace = data_handler.filter_stacktrace(
-      min_crash_stacktrace)
+  return {
+      'crash_type': min_state.crash_type,
+      'crash_address': min_state.crash_address,
+      'crash_state': min_state.crash_state,
+      'crash_stacktrace': data_handler.filter_stacktrace(min_crash_stacktrace),
+  }
+
+
+def _update_crash_result(testcase, crash_result_json):
```

Comment:
nit call last_crash_result_json and crash_result_json, crash_result_dict instead please. I think the fact it's JSON is a detail that minimize task doesn't need to know (as well readers of it).

Reply:
easy peasy!